server/protocols/jmap/src/test/java/org/apache/james/jmap/model/CreationMessageTest.java

```diff
@@ -159,4 +159,69 @@ public void mailboxIdShouldSetASingletonList() {
 
         assertThat(message.getMailboxIds()).containsExactly(mailboxId);
     }
+
+    @Test
+    public void isDraftWhenNoKeywordsSpecified() {
+        String mailboxId = "123";
+        CreationMessage message = CreationMessage.builder()
+            .mailboxId(mailboxId)
+            .build();
+
+        assertThat(message.isDraft()).isFalse();
+    }
+
+    @Test
+    public void isDraftShouldBeTrueWhenOldKeywordDraft() {
+        String mailboxId = "123";
+        CreationMessage message = CreationMessage.builder()
+            .mailboxId(mailboxId)
+            .isDraft(Optional.of(true))
+            .build();
+
+        assertThat(message.isDraft()).isTrue();
+    }
+
+    @Test
+    public void isDraftShouldBeFalseWhenOldKeywordNonDraft() {
+        String mailboxId = "123";
+        CreationMessage message = CreationMessage.builder()
+            .mailboxId(mailboxId)
+            .isAnswered(Optional.of(true))
```

Comment:
Do'nt you want ``isDraft(Optional.of(false))`` here instead ?

Reply:
Euh no... **OldKeyword** is present + That is not the isDraft flag => The message is not flagged.

Why should it be?

See below for other test cases.